zebra-network/src/protocol/external/inv.rs

```diff
@@ -81,3 +85,85 @@ impl ZcashDeserialize for InventoryHash {
         }
     }
 }
+
+const INV_HASH_SIZE: usize = 36;
+impl TrustedPreallocate for InventoryHash {
+    fn max_allocation() -> u64 {
+        // An Inventory hash takes 36 bytes, and we reserve at least one byte for the Vector length
+        // so we can never receive more than ((MAX_PROTOCOL_MESSAGE_LEN - 1) / 36) in a single message
+        ((MAX_PROTOCOL_MESSAGE_LEN - 1) / INV_HASH_SIZE) as u64
+    }
+}
+
+#[cfg(test)]
+mod test_trusted_preallocate {
+    use std::convert::TryInto;
+
+    use super::{InventoryHash, INV_HASH_SIZE, MAX_PROTOCOL_MESSAGE_LEN};
+    use zebra_chain::{
+        block,
+        serialization::{TrustedPreallocate, ZcashSerialize},
+        transaction,
+    };
+    #[test]
+    /// Confirm that each InventoryHash takes exactly INV_HASH_SIZE bytes when serialized.
+    /// This verifies that our calculated `TrustedPreallocate::max_allocation()` is indeed an upper bound.
+    fn inv_hash_size_is_correct() {
```

Comment:
Similarly, the inv tests should be proptests, in case we add more variants in future.